### ISSUE TITLE
Query function $in not working

### ISSUE THREAD

**Issue by mkholt** [delegateas/XrmDefinitelyTyped#168]
**Describe the bug**
When using retrieveMultiple filtering, the query function $in does not parse the property name correctly.

**To Reproduce**
1. Call the function ``Filter.$in("dg_number", [1,2])``
2. Notice that it outputs that filter ``Microsoft.Dynamics.CRM.In(PropertyName='',PropertyValues=[1,2])``

**Expected behavior**
The query function should contain the name of the property passed.

**Examples**
```
Filter.$in("dg_number", [1, 2])
"Microsoft.Dynamics.CRM.In(PropertyName='',PropertyValues=[1,2])"
Filter.$in("dgnumber", [1, 2])
"Microsoft.Dynamics.CRM.In(PropertyName=''dgnumber'',PropertyValues=[1,2])"
Filter.$in("_dg_number_value", [1, 2])
"Microsoft.Dynamics.CRM.In(PropertyName='dg_number_',PropertyValues=[1,2])"
```

**Environment**
- CRM/D365/CDS version: n/a
- Tool version: 4.4.3

**Comment by TomMalow** [comment]
You have to use `Filter.In` filter like this:
```typescript
XrmQuery.retrieveMultiple(x => x.accounts)
    .filter(x => Filter.$in(x.accountnumber,["12345"]))
```

**Comment by TomMalow** [comment]
After looking at it further we can see that we have an issue when using query function on fields that are not lookups and contains "_" in the name. Which for some strange reason is quite common!

**Comment by magesoe** [comment]
Fixed in 5.1.5, can you confirm it fixes it?

**Comment by mkholt** [comment]
Confirmed fixed in 5.1.5
